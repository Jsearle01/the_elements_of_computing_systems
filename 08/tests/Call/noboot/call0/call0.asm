@return_0
D=A
@SP
M=M+1
A=M-1
M=D
@LCL
D=M
@SP
M=M+1
A=M-1
M=D
@ARG
D=M
@SP
M=M+1
A=M-1
M=D
@THIS
D=M
@SP
M=M+1
A=M-1
M=D
@THAT
D=M
@SP
M=M+1
A=M-1
M=D
@SP
D=M
@ARG
M=D
@11
D=A
@ARG
M=M-D
@SP
D=M
@LCL
M=D
@testFunc
0;JMP
(return_0)
(testFunc)
