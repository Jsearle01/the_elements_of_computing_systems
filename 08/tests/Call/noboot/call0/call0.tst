load call0.asm,
output-file call0.out,
compare-to call0.cmp,
output-list
    RAM[0]%D1.6.1
    RAM[1]%D1.6.1
    RAM[2]%D1.6.1
    RAM[257]%D1.6.1
    RAM[258]%D1.6.1
    RAM[259]%D1.6.1
    RAM[260]%D1.6.1
    RAM[261]%D1.6.1;

set RAM[0] 256,
set RAM[1] 101,
set RAM[2] 102,
set RAM[3] 103,
set RAM[4] 104,

repeat 100 {
  ticktock;
}

output;

