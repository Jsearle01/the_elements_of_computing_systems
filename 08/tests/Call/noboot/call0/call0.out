| RAM[0] | RAM[1] | RAM[2] |RAM[257]|RAM[258]|RAM[259]|RAM[260]|RAM[261]|
